service/resource/cloudformation/create.go

```diff
@@ -1,7 +1,67 @@
 package cloudformation
 
-import "context"
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/aws-operator/service/key"
+	"github.com/giantswarm/microerror"
+)
 
 func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	createChangeState, ok := createChange.(StackState)
+	if !ok {
+		return microerror.Mask(fmt.Errorf("unexpected type, expecting %T, got %T", createChangeState, createChange))
+	}
+
+	mainTemplate, err := getMainTemplateBody(customObject)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	stackInput := &awscloudformation.CreateStackInput{
+		StackName:        aws.String(createChangeState.Name),
+		TemplateBody:     aws.String(mainTemplate),
+		TimeoutInMinutes: aws.Int64(defaultCreationTimeout),
+	}
```

Comment:
IMO this is state computation which should not be done here when we apply the create change. Here we should only take the state that is given and apply it via the dependency which in this case is an AWS client.

Reply:
ok, i've moved it to `newCreateChange`, pls take another look